### ISSUE TITLE
sdltheoraplay: Unknown pixel format with `--rgb` or `--rgba` command line

### ISSUE THREAD

**Issue by icculus** [libsdl-org/sdl12-compat#46]
- install dev packages for libogg, libvorbis, and libtheora
- install dev package for SDL 1.2 (so we have sdl-config, headers, and a lib to link against)
- Find an Ogg Theora file to play.
- hg clone https://hg.icculus.org/icculus/theoraplay
- cd theoraplay/test
- ./maketest.sh
- LD_LIBRARY_PATH=/where/i/built/sdl12-compat ./sdltheoraplay --rgb /path/to/my/theora/video.ogv

Output:

```
[...other debug stuff...]
Shadow surface failed: Unknown pixel format
Initialization failed!
```

Haven't researched this further yet. `--rgba` does this too.

**Comment by icculus** [comment]
This is actually a bug in theoraplay's test app, but this definitely didn't fail on SDL 1.2:

```c
const int alpha = (vidfmt == THEORAPLAY_VIDFMT_RGBA);
const int bits = 24 + (alpha * 8);
const Uint32 rmask = SDL_SwapLE32(0xFF0000FF);
const Uint32 gmask = SDL_SwapLE32(0x0000FF00);
const Uint32 bmask = SDL_SwapLE32(0x00FF0000);
const Uint32 amask = 0x00000000;
```

The `rmask` line has a bogus mask; it should be 0x000000FF. Fixing that error allows this to run under sdl12-compat, but we should look into why SDL 1.2 allowed this.

**Comment by icculus** [comment]
Pixel format created by SDL 1.2's SDL_CreateRGBSurface()...

```
(gdb) p/x *surface->format
$8 = {
  palette = 0x0, 
  BitsPerPixel = 0x20, 
  BytesPerPixel = 0x4, 
  Rloss = 0x0, 
  Gloss = 0x0, 
  Bloss = 0x0, 
  Aloss = 0x8, 
  Rshift = 0x0, 
  Gshift = 0x8, 
  Bshift = 0x10, 
  Ashift = 0x0, 
  Rmask = 0xff0000ff, 
  Gmask = 0xff00, 
  Bmask = 0xff0000, 
  Amask = 0x0, 
  colorkey = 0x0, 
  alpha = 0xff
}
```

As you can see, it kept the Rmask of 0xFF0000FF.

sdl12-compat calls into the SDL2 implementation of SDL_CreateRGBSurface and fails immediately [at this line](https://github.com/libsdl-org/SDL/blob/c0166a29b71fa1f5198e94c2a6da4224c2a2009a/src/video/SDL_surface.c#L163-L167) with SDL_PIXELFORMAT_UNKNOWN:

```c
    format = SDL_MasksToPixelFormatEnum(depth, Rmask, Gmask, Bmask, Amask);
    if (format == SDL_PIXELFORMAT_UNKNOWN) {
        SDL_SetError("Unknown pixel format");
```

I'm not sure the right way to protect against this sort of thing is. This renders the video correctly with SDL 1.2, so it must be ignoring that weird mask during blitting, but what is the algorithm to clean up this mask correctly before feeding it to SDL2?

**Comment by icculus** [comment]
On SDL 1.2, this falls into BlitNtoN, which is the generic cover-everything slow path, where it shrugs and decides both surfaces are 32 bit, so just copy it byte-for-byte...so it works by accident. I dunno, I think I'm going to close this as WONTFIX until it becomes a serious problem.

**Comment by icculus** [comment]
Sort of accidentally fixed this as a sort of special case.
